src/cpp/src/sampler.cpp

```diff
@@ -756,8 +756,7 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
             continue;
 
         size_t num_running_sequences = sequence_group->num_running_seqs();
-        size_t actual_seq_len = sequence_group->get_num_scheduled_tokens(); // points to a token which needs to be sampled
-        size_t padded_amount_of_processed_tokens = std::max(actual_seq_len, batch_seq_len);
+        size_t actual_seq_len = sequence_group->get_seq_len_to_sample();
```

Comment:
```suggestion
        size_t output_seq_len = sequence_group->get_output_seq_len();
```
IMO, it's a better name as by this function we define a number of tokens in last matmul